diff --git a/rpi4-ws/README.md b/rpi4-ws/README.md
index 0fe0a9c1ff9c..b9d044c70a69 100644
--- a/rpi4-ws/README.md
+++ b/rpi4-ws/README.md
@@ -509,6 +509,15 @@ After Linux finishes booting run xtest as follows:
 xtest -t regression
 ```
 
+If you want to, you can enable internet access with this command:
+
+```sh
+udhcpc -i eth0
+```
+
+This will allow you to access the internet (make sure that the ethernet cable
+is plugged in).
+
 ### Demo 2
 This demo instantiates a Linux VM and two OP-TEE VMs.
 ``` sh
diff --git a/rpi4-ws/configs/rpi4-single-vTEE/config.c b/rpi4-ws/configs/rpi4-single-vTEE/config.c
index bd85e2c5275c..82788fa812ca 100644
--- a/rpi4-ws/configs/rpi4-single-vTEE/config.c
+++ b/rpi4-ws/configs/rpi4-single-vTEE/config.c
@@ -33,7 +33,7 @@ struct vm_config linux = {
                 .shmem_id = 0,
             },
         },
-        .dev_num = 4,
+        .dev_num = 5,
         .devs =  (struct dev_region[]) {
             {
                 .pa   = 0xfc000000,
@@ -48,7 +48,7 @@ struct vm_config linux = {
 
             },
             {
-                .interrupt_num = 184,
+                .interrupt_num = 182,
                 .interrupts = (irqid_t[]) {
                     32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46,
                     47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61,
@@ -59,7 +59,7 @@ struct vm_config linux = {
                     117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128,
                     129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140,
                     141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152,
-                    153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164,
+                    153, 154, 155, 156, 159, 160, 161, 162, 163, 164,
                     165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176,
                     177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188,
                     189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200,
@@ -67,6 +67,14 @@ struct vm_config linux = {
                     213, 214, 215, 216, 
                 }
             },
+            {
+                /* Wired Network Device */
+                .pa = 0x7d580000, // Physical address of the genet device
+                .va = 0x7d580000, // Virtual address for the VM
+                .size = 0x10000,  // Size of the device memory region
+                .interrupt_num = 4,
+                .interrupts = (irqid_t[]) {0, 4, 157, 158} // Interrupts for the network device
+            },
             {
                 /* Arch timer interrupt */
                 .interrupt_num = 1,
diff --git a/support/linux-aarch64.config b/support/linux-aarch64.config
index 4f017686554c..2adabf6894a6 100644
--- a/support/linux-aarch64.config
+++ b/support/linux-aarch64.config
@@ -2360,7 +2360,7 @@ CONFIG_NET_VENDOR_AURORA=y
 # CONFIG_AURORA_NB8800 is not set
 CONFIG_NET_VENDOR_BROADCOM=y
 # CONFIG_B44 is not set
-CONFIG_BCMGENET=m
+CONFIG_BCMGENET=y
 # CONFIG_BNX2 is not set
 # CONFIG_CNIC is not set
 # CONFIG_TIGON3 is not set
