diff --git a/CROSSCON-Hypervisor-and-TEE-Isolation-Demos/buildroot/package/wolfssl/wolfssl.mk b/tmp/crosscon-demos/buildroot/package/wolfssl/wolfssl.mk
index cd3acd9411c8..17c06178a6b2 100644
--- a/CROSSCON-Hypervisor-and-TEE-Isolation-Demos/buildroot/package/wolfssl/wolfssl.mk
+++ b/tmp/crosscon-demos/buildroot/package/wolfssl/wolfssl.mk
@@ -32,17 +32,15 @@ else
 WOLFSSL_CONF_OPTS += --disable-sslv3
 endif

-# enable ARMv8 hardware acceleration
-ifeq ($(BR2_ARM_CPU_ARMV8A),y)
-WOLFSSL_CONF_OPTS += --enable-armasm
-# the flag -mstrict-align is needed to prevent build errors caused by
-# some inline assembly in parts of the AES structure using the "m"
-# constraint
-ifeq ($(BR2_aarch64),y)
-WOLFSSL_CONF_ENV += CPPFLAGS="$(TARGET_CPPFLAGS) -mstrict-align"
-endif
-else
+################################################################
+# CUSTOM - make wolfssl build properly for mtls                #
+# These setting ensure we don't get illegal instruction errors #
+################################################################
+
+# Enable pkcs#11
+WOLFSSL_CONF_OPTS += --enable-pkcs11
+
+# Control ARMv8 HW acceleration manually
 WOLFSSL_CONF_OPTS += --disable-armasm
-endif

 $(eval $(autotools-package))
