--- ../CROSSCON-Hypervisor-and-TEE-Isolation-Demos/buildroot/package/opensc/opensc.mk	2023-01-18 09:04:16.000000000 +0100
+++ ../tmp/crosscon-demos/buildroot/package/opensc/opensc.mk	2025-05-07 14:59:38.774907244 +0200
@@ -11,6 +11,6 @@
 OPENSC_CPE_ID_VENDOR = opensc_project
 OPENSC_DEPENDENCIES = openssl pcsc-lite
 OPENSC_INSTALL_STAGING = YES
-OPENSC_CONF_OPTS = --disable-cmocka --disable-strict
+OPENSC_CONF_OPTS = --disable-cmocka --disable-strict --enable-openssl
 
 $(eval $(autotools-package))
